File: flutter/ios/Classes/KmpblePlugin.swift
import Flutter
import UIKit
import KMPBLE

public class KmpblePlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "kmpble", binaryMessenger: registrar.messenger())
    let instance = KmpblePlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
    case "getPlatformInfoPeripheral":
      val peripheral = Peripheral()
      result.success(peripheral.getPlatformInfoPeripheral())
    case "getPlatformInfoScanner":
      val scanner = Scanner()
      result.success(scanner.getPlatformInfoScanner())
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}

File: flutter/android/src/main/kotlin/com/sherlockblue/kmpble/KmpblePlugin.kt
package com.sherlockblue.kmpble

import androidx.annotation.NonNull

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import com.sherlockblue.kmpble.Peripheral
import com.sherlockblue.kmpble.Scanner

/** KmpblePlugin */
class KmpblePlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "kmpble")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(call: MethodCall, result: Result) {
    when (call.method) {
      "getPlatformInfoPeripheral" -> {
        val peripheral = Peripheral()
        result.success(peripheral.getPlatformInfoPeripheral())
      }
      "getPlatformInfoScanner" -> {
        val scanner = Scanner()
        result.success(scanner.getPlatformInfoScanner())
      }
      "getPlatformVersion" -> {
        result.success("Android ${android.os.Build.VERSION.RELEASE}")
      }
      else -> result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}
